metricflow/naming/object_builder_scheme.py

```diff
@@ -0,0 +1,200 @@
+            len(call_parameter_sets.dimension_call_parameter_sets)
+            + len(call_parameter_sets.time_dimension_call_parameter_sets)
+            + len(call_parameter_sets.entity_call_parameter_sets)
+        )
+        if num_parameter_sets != 1:
+            raise ValueError(f"Did not find exactly 1 call parameter set. Got: {num_parameter_sets}")
+
+        for dimension_call_parameter_set in call_parameter_sets.dimension_call_parameter_sets:
+            return DimensionPattern(
+                EntityLinkPatternParameterSet(
+                    element_name=dimension_call_parameter_set.dimension_reference.element_name,
+                    entity_links=dimension_call_parameter_set.entity_path,
+                    time_granularity=None,
+                    date_part=None,
+                    fields_to_compare=(
+                        ParameterSetField.ELEMENT_NAME,
+                        ParameterSetField.ENTITY_LINKS,
+                        ParameterSetField.DATE_PART,
+                    ),
+                )
+            )
+
+        for time_dimension_call_parameter_set in call_parameter_sets.time_dimension_call_parameter_sets:
+            fields_to_compare = [
+                ParameterSetField.ELEMENT_NAME,
+                ParameterSetField.ENTITY_LINKS,
+                ParameterSetField.DATE_PART,
+            ]
+
+            if time_dimension_call_parameter_set.time_granularity is not None:
+                fields_to_compare.append(ParameterSetField.TIME_GRANULARITY)
+
+            return TimeDimensionPattern(
+                EntityLinkPatternParameterSet(
+                    element_name=time_dimension_call_parameter_set.time_dimension_reference.element_name,
+                    entity_links=time_dimension_call_parameter_set.entity_path,
+                    time_granularity=time_dimension_call_parameter_set.time_granularity,
+                    date_part=time_dimension_call_parameter_set.date_part,
+                    fields_to_compare=tuple(fields_to_compare),
+                )
+            )
+
+        for entity_call_parameter_set in call_parameter_sets.entity_call_parameter_sets:
+            return EntityLinkPattern(
+                EntityLinkPatternParameterSet(
+                    element_name=entity_call_parameter_set.entity_reference.element_name,
+                    entity_links=entity_call_parameter_set.entity_path,
+                    time_granularity=None,
+                    date_part=None,
+                    fields_to_compare=(
+                        ParameterSetField.ELEMENT_NAME,
+                        ParameterSetField.ENTITY_LINKS,
+                    ),
+                )
+            )
+
+        raise RuntimeError("There should have been a return associated with one of the CallParameterSets.")
+
+    @override
+    def input_str_follows_scheme(self, input_str: str) -> bool:
+        if ObjectBuilderNamingScheme._NAME_REGEX.match(input_str) is None:
```

Comment:
Do we need this regex, or can we jump to the parsing? It's not clear to me what the regex is adding, honestly.